arbor/include/arbor/util/optional.hpp

```diff
@@ -45,6 +43,9 @@
 namespace arb {
 namespace util {
 
+struct in_place_t {};
+static constexpr in_place_t in_place{};
```

Comment:
Should this go to `meta.hpp`, given that in theory it can be used by other types, like `any`?

Mind, you could just use the C++17 equivalent now that we have upgraded...

Reply:
Yep, we'll get C++17 PR in first, and then I'll update this PR to just use `std::in_place_t` instead.